corehq/toggles.py

```diff
@@ -125,6 +125,9 @@ def enabled_for_request(self, request):
         )
 
     def set(self, item, enabled, namespace=None):
+        if namespace == NAMESPACE_USER:
+            namespace = None  # because:
+            #     __init__() ... self.namespaces = [None if n == NAMESPACE_USER else n for n in namespaces]
```

Comment:
no, the way the toggle record is resolved is currently broken, if you pass `NAMESPACE_USER` into `enabled` `namespace` is set to `None`, but if you set it with `namespace` specified, then `user:` is appended...but there's no state right now in `enabled` which triggers that.

Reply:
Sorry, my comment may have been misleading because my code snippet pointed to a line in `enabled()`, but I'm talking about what's happening in `set()`.

Here are some scenarios inside `<toggle>.set('test@test.com', NAMESPACE_USER)`:
```py
>>> from toggle.shortcuts import namespaced_item
>>> NAMESPACE_USER = 'user'

# the old way would have stored this in the database
>>> namespaced_item("test@test.com", NAMESPACE_USER)
'user:test@test.com'

# what your new code would store in the database
>>> namespaced_item("test@test.com", None)
'test@test.com'
```

This looks wrong to me as there will be no distinction between user and domain toggles. The toggle code is terribly confusing and hard for me to follow. Maybe I'm still not understanding what's going on. However, this feels like a dangerous change.

Reply:
and that being said, `enabled` will never (due to legacy reasons because toggles were originally for users without namespacing) pull up a toggle that has been set to `user:test@test.com` because the `NAMESPACE_USER` will always be set to none

see:
https://github.com/dimagi/commcare-hq/blob/master/corehq/toggles.py#L97-L99

Reply:
all users are stored in the `Toggle` doc without `user:` prepended. if it is...there's no way of getting that out in `enabled`

Reply:
it's super confusing! which is why it took me a bit to figure out wtf is going on LOL.

Reply:
offlined. I was confused, and this is fine.